src/test/java/com/revature/controllers/AddressControllerTest.java

```diff
@@ -0,0 +1,197 @@
+		
+		assertEquals(HttpStatus.NOT_FOUND.value(), response.getStatus());
+		assertNotEquals(this.jsonReviewList.write(expected).getJson(), response.getContentAsString());
+		verify(this.userv, times(1)).findById(1);
+		verify(this.aserv, times(0)).findUsersAddresses(dummyUser);
+	}
+	
+	@Test
+	void testGetUserAddress_NoAddress() throws Exception{
+		given(userv.findById(1)).willReturn(Optional.of(dummyUser));
+		MockHttpServletRequestBuilder request = get(MAPPING + "/1").accept(MediaType.APPLICATION_JSON);
+		MockHttpServletResponse response = mvc.perform(request).andReturn().getResponse();
+		
+		assertEquals(HttpStatus.OK.value(), response.getStatus());
+		assertNotEquals("<>", response.getContentAsString());
+		verify(this.userv, times(1)).findById(1);
+		verify(this.aserv, times(1)).findUsersAddresses(dummyUser);
+	}
+	
+	@Test
+	void testUpdateAddress_Succeed() throws Exception {
+		AddressRequest newReq = new AddressRequest();
+		Set<Address> addresses = new HashSet<Address>();
+		addresses.add(dummyAddress);
+		dummyUser.setAddresses(addresses);
+		
+		newReq.setStreet("105 maybarry");
+		newReq.setSecondary("");
+		newReq.setCity("Far Far Away");
+		newReq.setZip("70530");
+		newReq.setState("NJ");
+		
+		given(aserv.update(newReq, 1, dummyUser)).willReturn(dummyAddress);
+
+		String jsonContent = this.JsonAddressRequest.write(newReq).getJson();
+		
+		MockHttpServletRequestBuilder request = put(MAPPING + "/1").contentType(MediaType.APPLICATION_JSON)
+				.content(jsonContent).sessionAttr("user", dummyUser);
+		MockHttpServletResponse response = mvc.perform(request).andReturn().getResponse();
+		
+		assertEquals(HttpStatus.OK.value(), response.getStatus());
+		assertEquals(this.jsonAddress.write(dummyAddress).getJson(), response.getContentAsString());
+		verify(this.aserv, times(1)).update(newReq, 1, dummyUser);
+	}
+	
+	@Test
+	void testUpdateAddress_Fail_NoAddresses() throws Exception{
+		AddressRequest newReq = new AddressRequest();
+		Set<Address> addresses = new HashSet<Address>();
+		addresses.add(dummyAddress);
+		
+		newReq.setStreet("105 maybarry");
+		newReq.setSecondary("");
+		newReq.setCity("Far Far Away");
+		newReq.setZip("70530");
+		newReq.setState("NJ");
+		
+		String jsonContent = this.JsonAddressRequest.write(newReq).getJson();
+		
+		MockHttpServletRequestBuilder request = put(MAPPING + "/1").contentType(MediaType.APPLICATION_JSON)
+				.content(jsonContent).sessionAttr("user", dummyUser);
+		MockHttpServletResponse response = mvc.perform(request).andReturn().getResponse();
+		
+		assertEquals(HttpStatus.OK.value(), response.getStatus());
+		assertEquals("", response.getContentAsString());
+		verify(this.aserv, times(1)).update(newReq, 1, dummyUser);
+	}
+	
+	
+	@Test
+	void testAddAddress_Success() throws Exception{
+		AddressRequest newReq = new AddressRequest();
+		newReq.setStreet("105 maybarry");
+		newReq.setSecondary("");
+		newReq.setCity("Far Far Away");
+		newReq.setZip("70530");
+		newReq.setState("NJ");
```

Comment:
These should match the values of those in `dummyAddress`, e.g., `dummyAddress.getStreet()` instead of `"105 maybarry"` because you're inserting a new `Address` whole field values are exactly taken from `newReq`